src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -72,22 +70,12 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
             //          installs a service roslyn visual studio workspace requires
             shell.LoadPackage(Guids.RoslynPackageId, out var setupPackage);
 
-            _miscellaneousFilesWorkspace = this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
-            if (_miscellaneousFilesWorkspace != null)
-            {
-                // make sure solution crawler start once everything has been setup.
-                _miscellaneousFilesWorkspace.StartSolutionCrawler();
-            }
-
-            RegisterMiscellaneousFilesWorkspaceInformation(_miscellaneousFilesWorkspace);
+            var miscellaneousFilesWorkspace = this.ComponentModel.GetService<MiscellaneousFilesWorkspace>();
+            RegisterMiscellaneousFilesWorkspaceInformation(miscellaneousFilesWorkspace);
 
             this.Workspace = this.CreateWorkspace();
             if (await IsInIdeModeAsync(this.Workspace, cancellationToken).ConfigureAwait(true))
             {
-                // make sure solution crawler start once everything has been setup.
-                // this also should be started before any of workspace events start firing
-                this.Workspace.StartSolutionCrawler();
-
                 // start remote host
```

Comment:
unfortunately, that wasn't possible since a service in this package initialize (I think nuget package one ) required OOP before workspace fire first events. either how remote host started or nuget package needs to move to Listener model as well to completely remove this.

Reply:
https://github.com/dotnet/roslyn/issues/36559